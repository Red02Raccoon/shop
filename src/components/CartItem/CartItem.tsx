import React from 'react';

import { IProductItem } from '../../common/types';

const CartItem: React.FC<IProductItem> = ({ id, title, image, price }) => {
    return (
        <div className="flex py-6">
            <div className="h-24 w-24 flex-shrink-0 overflow-hidden rounded-md border border-gray-200">
                <img src={image} className="h-full w-full object-cover object-center" alt="test" />
            </div>

            <div className="ml-4 flex flex-1 flex-col">
                <div>
                    <div className="flex justify-between text-base font-medium text-gray-900">
                        <h3>
                            <span> {title} </span>
                        </h3>
                        <p className="ml-4">{price}</p>
                    </div>
                </div>
                <div className="flex flex-1 items-end justify-between text-sm">
                    <p className="text-gray-500">Qty 1</p>

                    <div className="flex">
                        <button
                            type="button"
                            className="font-medium text-indigo-600 hover:text-indigo-500"
                            data-testid="remove-button"
                        >
                            Remove
                        </button>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default CartItem;
