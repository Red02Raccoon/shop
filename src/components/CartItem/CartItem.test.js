import { render, screen } from "@testing-library/react";

import CartItem from "./CartItem";

//NOTE: learn different assertions and selectors
describe("CartItem", () => {
  it("should display remove button", () => {
    const props = {
      id: 12,
      title: "test",
      image: "/test/img-1.png",
      price: 12,
    };
    render(<CartItem {...props} />);

    const button = screen.getByRole("button", { name: "Remove" });
    expect(button).toBeInTheDocument();

    const button2 = screen.getByRole("button");
    expect(button2).toHaveTextContent("Remove");

    const button3 = screen.getByTestId("remove-button");
    expect(button3).toBeVisible();
  });

  it("should match the snapshot", () => {
    const props = {
      id: 12,
      title: "test",
      image: "/test/img-1.png",
      price: 12,
    };
    const { asFragment } = render(<CartItem {...props} />);

    expect(asFragment(<CartItem {...props} />)).toMatchSnapshot();
  });

  it("should", () => {
    const props = {
      id: 12,
      title: "test",
      image: "/test/img-1.png",
      price: 12,
    };
    const { container, getByTestId } = render(<CartItem {...props} />);

    console.log(container.firstChild);

    expect(getByTestId("remove-button")).toBeInTheDocument();
  });
});
