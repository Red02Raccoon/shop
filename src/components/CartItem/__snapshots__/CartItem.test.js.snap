// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex py-6"
  >
    <div
      class="h-24 w-24 flex-shrink-0 overflow-hidden rounded-md border border-gray-200"
    >
      <img
        alt="test"
        class="h-full w-full object-cover object-center"
        src="/test/img-1.png"
      />
    </div>
    <div
      class="ml-4 flex flex-1 flex-col"
    >
      <div>
        <div
          class="flex justify-between text-base font-medium text-gray-900"
        >
          <h3>
            <span>
               test 
            </span>
          </h3>
          <p
            class="ml-4"
          >
            12
          </p>
        </div>
      </div>
      <div
        class="flex flex-1 items-end justify-between text-sm"
      >
        <p
          class="text-gray-500"
        >
          Qty 1
        </p>
        <div
          class="flex"
        >
          <button
            class="font-medium text-indigo-600 hover:text-indigo-500"
            data-testid="remove-button"
            type="button"
          >
            Remove
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
